_summaries/bottou2010large_2.md

```diff
@@ -1,19 +1,126 @@
+
+$$
+E_{n}(f) = \frac{1}{n}\sum^{n}_{i=1}l(f(x_{i}), y_{i})
+$$
+
+The hope is that the Empirical Risk is a good approximation of the Expected Risk and thus reducing loss in the Empirical should give a reduction in loss on the Expected.
+
+_**How does Gradient Decent and Stochastic Gradient Decent work with Risks?**_
+
+Gradient Decent is able to reach the Empirical Optimum under sufficient assumptions on the learning rate, optimization, etc.
+
+SGD needs the same assumptions as Gradient Decent, but also needs decaying learning rates to cope with the noise in the gradient approximations.
+
+Because of these assumptions and the noise SGD makes, it can take longer for SGD to reach the Empirical Optimum.
+
+_**So... Why use SGD?**_
+
+When tasks are sufficiently large and computationally bound its usually better to get within some threshold of the Empirical Optimum
+
+$$
+E_{n}(\tilde{f}_{n}) < E_{n}(f_{n}) + p
+$$
+
+(where $$p$$ is the extra risk you are willing to accept to decrease training time)
+
+Now we have set a predefined accuracy that we want our algorithm to reach, the error between our predefined accuracy and the best possible accuracy we could ever get is called _Excess Error_
+
+$$
+\mathcal{E} = \mathbb{E} [E(\tilde{f}_{n}) - E(f^{*})]
+$$
+
+The authors found that SGD is quicker to reach a predefined Excess Error
+
+With sufficient training samples, you can model the Expected Risk (the actual distribution) more accurately, and be able to approach some predefined value of Expected Risk quicker with SGD than GD can.
+
+
+_**Does 2nd Order Stochastic Gradient Decent (2SGD) have any benefits?**_
+
+2SGD, with sufficient assumptions (convexity and regularity), will approach optimum solution as quickly as possible
+
+However, 2SGD requires a large dense matrix that is too computationally expense to compute
+
+In order to overcome the matrix, SGDQN and Averaged Stochastic Gradient Descent (ASGD) attempt to workaround it.
+
+Both SQDQN and ASGD have nice properies and can converge faster than SGD, but still come at a significant cost (ASGD requires 2 weight matrices)
+
+
+_**How well does the paper perform?**_
+
+SGD is used quite often now, so it did become quite popular
+
+The authors tested SGD, ASGD, and SGDQN on various types of datasets showing that Stochastic Gradient Decent is capable of the Expected Risk quickly beating out other algorithms (30s -> 2.3s)
+
+//TODO include plots here
+
+
+_**Interesting Variants**_
+
+The authors further break down the Excess Error term into
+- Approximation Error (how many functions can you search over for the optimum)
+- Estimation Error (how well does the training set represent the real distribution)
+- Optimization Error (did you stop the optimization too early, or set $$p$$ too high)
+
+You can then choose, specific to your problem, which of these risks to minimize
+
+In some cases GD may be the better option than SGD if you are working on a smaller machine learning problem.
+
 
 ## TL;DR
```

Comment:
I think you did a good job of explaining the differences between GD and SGD without going too into mathematics. In section 3, the paper goes heavy into math and it might not necessarily belong in a summary.

Reply:
Yeah, I agree, the paper felt more like a very deep proof.  And as cool as that is, I don't think you have to really remember it (it's more like, okay this is the right conclusion to come to).

That being said though if you think I need to add any other equations totally let me know, I tried to only pull the important ones.